### PR TITLE
GUI to enter phone number #2

### PR DESCRIPTION
**Pull request by eniac111**
Please, review it. It has a conflict with your changes.

### CODE REVIEW

File: appinfo/application.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+class Application extends App {
+	public function __construct (array $urlParams = array()) {
+		parent::__construct('richdocuments', $urlParams);
+
+		$container = $this->getContainer();
+
+		/**
+		 * Controllers
+		 */
+		$container->registerService('SettingsController', function($c) {
```
Comment (ChristophWurst) [inline appinfo/application.php]: this shouldn't be necessary. You can remove that

File: appinfo/application.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			return new SettingsController(
+				$c->query('AppName'),
+				$c->query('L10N'),
+				$c->query('AppConfig'),
+				$c->query('UserId')
+			);
+		});
+
+		$container->registerService('AppConfig', function($c) {
```
Comment (ChristophWurst) [inline appinfo/application.php]: same here

File: appinfo/application.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			return new AppConfig(
+				$c->query('CoreConfig')
+			);
+		});
+
+		/**
+		 * Core
+		 */
+		$container->registerService('CoreConfig', function($c) {
```
Comment (ChristophWurst) [inline appinfo/application.php]: same with the service below

File: controller/settingscontroller.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+//use OCA\Richdocuments\Filter;
+
+class SettingsController extends Controller{
+
+	private $userId;
+	private $l10n;
+	private $appConfig;
+  private $appName = 'TwoFactor_Sms'
+
+	public function __construct($appName, IRequest $request, IL10N $l10n, AppConfig $appConfig, $userId){
```
Comment (ChristophWurst) [inline controller/settingscontroller.php]: change ``AppConfig`` to ``IConfig`` and ``$useId`` to ``$UserId``, then dependency injection should be able to create the controller automatically.

File: controller/settingscontroller.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+namespace OCA\TwoFactor_Sms\Controller;
+
+use \OCP\AppFramework\Controller;
+use \OCP\IConfig;
+use \OCP\IRequest;
+use \OCP\IL10N;
+use \OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\TemplateResponse;
+
+//use OCA\Richdocuments\AppConfig;
```
Comment (ChristophWurst) [inline controller/settingscontroller.php]: remove this

File: controller/settingscontroller.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+			$this->appName,
+			'personal',
+			[ 'phone' => $this->appConfig->getUserValue($this->userId, 'phone') ],
+			'blank'
+		);
+	}
+
+	public function saveSettings($phone){
+		if (is_null($phone)){
+      $response = array(
```
Comment (ChristophWurst) [inline controller/settingscontroller.php]: use ``[]`` array syntax and fix indentation. Returning the array directly makes the code easier to read then storing it into a temporary variable

File: js/settings.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+                        OC.msg.startSaving('#twofactor-sms-personal .msg');
+                        $.post(OC.filePath('twofactor_sms', 'ajax', 'settings/personal.php'), data, smsSettings.afterSave);
+                },
+                afterSave : function(data){
+                        OC.msg.finishedSaving('#twofactor-sms-personal .msg', data);
+                }
+        };
+        $('#phone-number').blur(smsSettings.save);
+        $('#phone-number').keypress(function( event ) {
+                                                if (event.which == 13) {
```
Comment (ChristophWurst) [inline js/settings.js]: use ``===``

File: js/settings.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+$(document).ready(function(){
+        var smsSettings = {
+                save : function() {
+                        var data = {
+                                savePath : $('#phone-number').val()
+                        };
+                        OC.msg.startSaving('#twofactor-sms-personal .msg');
+                        $.post(OC.filePath('twofactor_sms', 'ajax', 'settings/personal.php'), data, smsSettings.afterSave);
```
Comment (ChristophWurst) [inline js/settings.js]: any specific reason why you chose a ``/ajax/`` url? I'd prefer to remove that prefix

File: lib/AppInfo/Application.php [hunk 9]
```diff
@@ -49,6 +49,8 @@ private function getSmsProviderClass($name) {
 		switch ($name) {
 			case 'websms.de':
 				return '\OCA\TwoFactor_Sms\Service\SmsProvider\WebSmsDe';
+			// case 'mail2sms':
```
Comment (ChristophWurst) [inline lib/AppInfo/Application.php]: remove if not needed

File: lib/appconfig.php [hunk 10]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace OCA\TwoFactor_Sms;
+
+use \OCP\IConfig;
+
+ class AppConfig{
```
Comment (ChristophWurst) [inline lib/appconfig.php]: do we really need this class?

**Comment by ChristophWurst** [comment]
Hi @eniac111,

thanks a lot for your PR. I'm sorry that it took so long until I reviewed it, apparently GitHub did not notify me about this.
Feel free to push your branch to this repo directly, then we can finish this together. I've added you as collaborator, you should receive an invitation soon.

**Comment by eniac111** [comment]
Hello @ChristophWurst ,

Thank you for the invitation and the review. I'll take some time for this after Thursday.

**Comment by ChristophWurst** [comment]
Hey @eniac111,

what's the status here? :-)

**Comment by eniac111** [comment]
Hi @ChristophWurst :) 

Excuse me, I was a bit busy at work the last months. The code just have to be cleaned but I see a phone number field in the personal settings of the last versions of NextCloud. We can use it, so I'll inspect this. It's better to close this request for now.

Cheers,
Blago

**Comment by ChristophWurst** [comment]
> We can use it, so I'll inspect this

Right, that would make sense and simplify the UX.

Thanks

**Pull request closed without merging**